Sources/TuistLoader/Models+ManifestMappers/Headers+ManifestMapper.swift

```diff
@@ -12,58 +12,142 @@ extension TuistGraph.Headers {
+}
+
+extension FileListGlob {
+    fileprivate func unfold(
+        generatorPaths: GeneratorPaths,
+        basenameFilter: Set<String>?,
+        additionalExcluding: Set<AbsolutePath>
+    ) throws -> [AbsolutePath] {
+        let resolvedPath = try generatorPaths.resolve(path: glob)
+        let resolvedExcluding = (try resolvedExcluding(generatorPaths: generatorPaths)).union(additionalExcluding)
+        return resolvedPath.unfold(basenameFilter: basenameFilter, excluding: resolvedExcluding)
+    }
+
+    fileprivate func resolvedExcluding(generatorPaths: GeneratorPaths) throws -> Set<AbsolutePath> {
+        guard !excluding.isEmpty else { return [] }
+        var result: Set<AbsolutePath> = []
+        try excluding.forEach { path in
+            let resolved = try generatorPaths.resolve(path: path).pathString
+            let absolute = AbsolutePath(resolved)
+            let globs = AbsolutePath(absolute.dirname).glob(absolute.basename)
+            result.formUnion(globs)
+        }
+        return result
+    }
+}
+
+extension AbsolutePath {
+    fileprivate func unfold(basenameFilter: Set<String>?, excluding: Set<AbsolutePath>) -> [AbsolutePath] {
+        FileHandler.shared.glob(AbsolutePath.root, glob: String(pathString.dropFirst())).filter {
+            guard let fileExtension = $0.extension else {
+                return false
+            }
+            guard TuistGraph.Headers.extensions.contains(".\(fileExtension)"),
+                  !excluding.contains($0)
+            else {
+                return false
+            }
+            if let basenameFilter = basenameFilter {
+                return basenameFilter.contains($0.basename)
+            } else {
+                return true
+            }
+        }
+    }
+}
+
+extension ProjectDescription.Headers {
+    fileprivate enum Constants {
+        static let regex = try? NSRegularExpression(pattern: "(?<=^#import (\\/|\"|<))(.+\\.h)", options: [])
+        static let ignoreImports = ["UIKit", "Foundation"]
+    }
+
+    fileprivate func publicHeadersFromUmbrella(resolvedPath: AbsolutePath?) throws -> Set<String>? {
+        guard let resolvedPath = resolvedPath else {
+            return nil
+        }
+        let umbrellaContent = try String(contentsOf: resolvedPath.url, encoding: .utf8)
+        let lines = umbrellaContent.components(separatedBy: .newlines)
+
+        let foundHeaders: [String] = lines.compactMap { line in
+            let stripped = line.trimmingCharacters(in: .whitespaces)
+            guard stripped.hasPrefix("#import"),
+                  let found = Constants.regex?.fistMatch(in: stripped),
+                  !Constants.ignoreImports.contains(where: found.hasPrefix)
+            else {
+                return nil
+            }
+            return found.components(separatedBy: "/").last
+        }
+        return Set(foundHeaders)
+    }
+}
+
+extension NSRegularExpression {
+    fileprivate func fistMatch(in string: String) -> String? {
```

Comment:
I recommend extracting this one into a extension of `NSRegularExpression` in `TuistSupport` like we are doing with [other Foundation utilities](https://github.com/tuist/tuist/blob/main/Sources/TuistSupport/Extensions/String+Extras.swift).

Reply:
done